datadog.go

```diff
@@ -0,0 +1,158 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// Package datadog contains a Datadog exporter.
+//
+// This exporter is currently work in progress
+package datadog
+
+// import "go.opencensus.io/exporter/datadog"
+
+import (
+	"bytes"
+	"errors"
+	"log"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+// Exporter exports stats to Datadog
+type Exporter struct {
+	opts      Options
+	collector *collector
+}
+
+// Options contains options for configuring the exporter
+type Options struct {
+	// Namespace to prepend to all metrics
+	Namespace string
+
+	// Host for DogStatsD connection
+	Host string
+
+	// Port for DogStatsD connection
+	Port string
+
+	// OnError is the hook to be called when there is
+	// an error occurred when uploading the stats data.
+	// If no custom hook is set, errors are logged.
+	// Optional.
+	OnError func(err error)
+
+	// Tags are global tags added to each metric
+	Tags []string
+}
+
+const (
+	defaultHost   = "localhost"
+	defaultPort   = "8125"
+	opencensusTag = "source:Opencensus"
+)
+
+var (
+	newExporterOnce      sync.Once
+	errSingletonExporter = errors.New("expecting only one exporter per instance")
+	err                  = errSingletonExporter
+	exporter             *Exporter
+	tags                 = []string{opencensusTag}
+)
+
+// NewExporter returns an exporter that exports stats to Datadog
+func NewExporter(o Options) (*Exporter, error) {
+	newExporterOnce.Do(func() {
+		exporter, err = newExporter(o)
+	})
+	return exporter, err
+}
+
+func (o *Options) getEndpoint() string {
+	var endpoint string
+	host, port := o.Host, o.Port
+	if host == "" || port == "" {
+		host, port = defaultHost, defaultPort
+	}
+	endpoint = host + ":" + port
```

Comment:
you can do `endpoint := host + ":" + port` and save the `var` declaration at the top. Also this function deserves docstrings.